lasy/profiles/from_openpmd_profile.py

```diff
@@ -0,0 +1,129 @@
+        If not, the envelope is obtained from the electric field
+        using a Hilbert transform
+
+    prefix : string
+        Prefix of the openPMD file from which the envelope is red.
+        Only used when envelope=True.
+        The provided iteration is read from <path>/<prefix>_%T.h5.
+
+    wavelength : float (in meter)
+        The main laser wavelength :math:`\\lambda_0` of the laser, which defines
+        :math:`\\omega_0`, according to :math:`\\omega_0 = 2\\pi c/\\lambda_0`.
+        This argument is optional, and will overwrite the following default
+        behavior:
+        - if envelope is True, the central wavelength is read from the openPMD
+        file at openPMD envelope standard.
+        - if envelope is False, the central wavelength is obtained from the
+        Hilbert transform.
+    """
+
+    def __init__(
+        self,
+        path,
+        iteration,
+        pol,
+        field,
+        coord=None,
+        envelope=False,
+        prefix=None,
+        wavelength=None,
+    ):
+        ts = OpenPMDTimeSeries(path)
+        F, m = ts.get_field(iteration=iteration, field=field, coord=coord, theta=None)
+        assert m.axes in [
+            {0: "x", 1: "y", 2: "z"},
+            {0: "z", 1: "y", 2: "x"},
+            {0: "x", 1: "y", 2: "t"},
+            {0: "t", 1: "y", 2: "x"},
+        ]
+
+        if m.axes in [{0: "z", 1: "y", 2: "x"}, {0: "t", 1: "y", 2: "x"}]:
+            F = F.swapaxes(0, 2)
+
+        if "z" in m.axes.values():
+            dt = m.dz / c
+            t = (m.z - m.z[0]) / c
+        else:
+            dt = m.dt
+            t = m.t
+        axes = {"x": m.x, "y": m.y, "t": t}
+
+        # If array does not contain the envelope but the electric field,
+        # extract the envelope with a Hilbert transform
+        if envelope == False:
+            # Assumes z is last dimension!
+            h = hilbert(F)
+
+            if "z" in m.axes.values():
+                # Flip to get complex envelope in t assuming z = -c*t
+                h = np.flip(h, axis=2)
+
+            # Get central wavelength from array
+            phase = np.unwrap(np.angle(h))
+            omg0_h = (
+                -np.average(
+                    np.diff(phase),
+                    weights=0.5 * (np.abs(h[:, :, 1:]) + np.abs(h[:, :, :-1])),
+                )
+                / dt
+            )
+            wavelength_loc = 2 * np.pi * c / omg0_h
+            if wavelength is not None:
+                wavelength_loc = wavelength
+            array = h * np.exp(1j * omg0_h * t)
+        else:
+            if wavelength is None:
+                s = io.Series(path + "/" + prefix + "_%T.h5", io.Access.read_only)
+                it = s.iterations[iteration]
+                omg0 = it.meshes["laserEnvelope"].get_attribute("angularFrequency")
+                wavelength_loc = 2 * np.pi * c / omg0
+            else:
+                wavelength_loc = wavelength
+            array = F
```

Comment:
maybe if wavelength  is provided and differs from the one in the field, we should physically  re-define the field with the new wavelength, e.g. 
```
            array = F
            s = io.Series(path + "/" + prefix + "_%T.h5", io.Access.read_only)
            it = s.iterations[iteration]
            omg0 = it.meshes["laserEnvelope"].get_attribute("angularFrequency")

            if wavelength is None:
                wavelength_loc = 2 * np.pi * c / omg0
            else:
                wavelength_loc = wavelength
                array *= np.exp(1j * (2 * np.pi * c  / wavelength - omg0) * t)
```

Reply:
Yes, we could do sth like this, I would keep this for another PR, as there are quite a few options that could make sense. I put it int #156.

Reply:
likewise, discussed offline, this option is useful and independent from this PR.